device/MatrixMultiplication.cpp

```diff
@@ -237,16 +372,38 @@ void MatrixMultiplication(DramLine const *const a, DramLine const *const b, Dram
 // C, to make sure that the compiler doesn't try to look for dependencies/conflicts
 #pragma HLS INTERFACE m_axi offset = slave port = c_read bundle = c_read
 #pragma HLS INTERFACE m_axi offset = slave port = c_write bundle = c_write
+#pragma HLS INTERFACE s_axilite port = a
+#pragma HLS INTERFACE s_axilite port = b
+#pragma HLS INTERFACE s_axilite port = c_read
+#pragma HLS INTERFACE s_axilite port = c_write
+#pragma HLS INTERFACE s_axilite port = size_n
+#pragma HLS INTERFACE s_axilite port = size_k
+#pragma HLS INTERFACE s_axilite port = size_m
```

Comment:
I would expect that this interface shouldn't have any concept of addressing, strobes, etc. I think a streaming interface would be more suitable? I believe HLS calls these `axis` or `ap_fifo`. Also, are you sure that we realize the full data path width?

Reply:
[AXI4-Lite](https://www.xilinx.com/products/intellectual-property/axi_lite_ipif.html) is an interface specifically meant for passing smaller 32-bit arguments to a kernel, it's standard for the OpenCL interface. It's not a memory mapped interface. As far as I am aware, there's no other option if you want to be compatible with OpenCL!